apps/galaxy/src/services/signin/useSignIn.ts

```diff
@@ -0,0 +1,46 @@
+import axios from "../utils/axios";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import { useToast } from "@cc/ui-chakra";
+import { AxiosError } from "axios";
+
+export default function useSignIn() {
+  const [isLoading, setIsLoading] = useState(false);
+  const toast = useToast();
+  const router = useRouter();
+  const handleSignInSubmit = async (username: string, password: string) => {
+    setIsLoading(true);
+    try {
+      const response = await axios.post("/staff/signin", {
+        username: username,
+        password: password,
+      });
+      if (response.status === 200) {
+        router.push("/");
+      }
+    } catch (error) {
+      const err = error as AxiosError;
+      if (err.response?.status === 400) {
+        toast({
+          title: "Sign in failed",
+          description: "Username and password is not authenticated",
+          status: "error",
+          duration: 6000,
```

Comment:
I remember there was an idea to turn this into a custom util hook, how did that go?

Reply:
since there is only ` status: "error"` and `duration: 6000,` can be reused in a custom hook. After a discussion with Robert and Tato, we think using a custom hook is unnecessary.